core/src/main/java/edu/ucr/cs/riple/core/metadata/index/Bank.java

```diff
@@ -50,18 +52,22 @@
   /** Factory instance to parse outputs. */
   private final Factory<T> factory;
   /** Path to file where outputs are stored. */
-  private final Path path;
+  private final ImmutableSet<Path> paths;
 
-  public Bank(Path path, Factory<T> factory) {
+  public Bank(Stream<Path> paths, Factory<T> factory) {
```

Comment:
I personally find it uncomfortable to see `Stream` being passed around because they are on-time-use only. Also based on the current usages, it would be cleaner to accept a `ImmuableSet`.

Reply:
Thank you very much for noting that, I mistakenly assumed since they don't have `add` method, they are immutable. This also has some other problems as you mentioned. 706face8159aaeddaee2861e298432f9930aeacd